static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -1,159 +1,160 @@
+                buttons.forEach(function (button, index) {
+                    var checkbox = checkboxes[index];
+                    var eventHandler = buttonClickHandlers[button.id];
 
-            return fastdomPromise.write(function () {
-                if (firstButton) {
-                    firstButton.focus();
-                }
-                // Prevents scrolling on the body
-                html.css('overflow', 'hidden');
+                    checkbox.parentNode.replaceChild(button, checkbox);
+                    button.addEventListener('click', eventHandler);
+                });
             });
         });
     }
 
-    function animateMenuClose() {
-        return fastdomPromise.write(function () {
-            if (mainMenuEl.hasClass('shown')) {
-                mainMenuEl.addClass('off-screen');
-
-                veggieBurgerLink.removeClass('new-header__nav__menu-button--open');
-                veggieBurgerLink.attr('href', mainMenuId);
-
-                // TODO: Support browsers that don't have transitions
-                // We still want to hide this
-                if (mainMenuEl.length > 0) {
-
-                    mainMenuEl[0].addEventListener('transitionend', function handler() {
-
-                        mainMenuEl[0].removeEventListener('transitionend', handler);
-
-                        return fastdomPromise.write(function () {
-                            mainMenuEl.removeClass('off-screen');
-                            mainMenuEl.removeClass('shown');
-                        }).then(function () {
-                            return fastdomPromise.write(function () {
-                                var mainListItems = $('.main-navigation__item');
-                                // Remove possible ordering for the lists
-                                mainListItems.removeAttr('style');
-                                // No targetItem to put in as the parameter. All lists should close.
-                                closeAllOtherPrimaryLists();
-
-                                $('.new-header__nav__menu-button').focus();
-                                // Users should be able to scroll again
-                                html.css('overflow', '');
-                            });
-                        });
-                    });
+    function veggieBurgerClickHandler(event) {
+        var button = event.target;
+        var mainMenu = qwery('#main-menu')[0];
+
+        function menuIsOpen() {
+            return (button.getAttribute('aria-expanded') === 'true');
+        }
+
+        if (!mainMenu) {
+            return;
+        }
+        if (menuIsOpen()) {
+            fastdom.write(function () {
+                var mainListItems = qwery('.main-navigation__item');
+
+                button.setAttribute('aria-expanded', 'false');
+                mainMenu.setAttribute('aria-hidden', 'true');
+                veggieBurgerLink[0].classList.remove('new-header__nav__menu-button--open');
```

Comment:
Sure thing. I've also brought the variable assignment into the handler function as it is not used elsewhere